### DISCUSSION TITLE
3D printed Photobox incl. 3 Button Bluetooth Buzzer

### DISCUSSION THREAD

**Discussion by Striker62** [photobooth-app/photobooth-app#244] [Show and tell]
![IMG_7513](https://github.com/photobooth-app/photobooth-app/assets/37927465/7a62ae6f-f3c2-4119-999d-97d50ad0f907)

Thanks to mgineer85 .
My booth is ready for the action.
On top you see my 3 Button Bluetooth Buzzer

**Comment by PixelAlarm** [comment]
Really nice,
could you give some more Info about the buzzer :)
Thanks!

**Comment by Striker62** [comment]
sure,
basis was this https://github.com/photobooth-app/photobooth-buzzer
I extend the main.ccp to deal with 3 buttons.
I can share the file without any guaranty ;-)
I use a az-delivery "ESP32 Wroom modul" this comes with bluetooth. I use an small Powerbank that I got as a giveaway on a fair.
I designed my own Box for the Buttons  and 3D print it.

The only two things you have to do is activate bluetooth on the Rpi and pair the buzzer with the Rpi
and define 3 keyboard key in the admin section of the Photobooth.
I defined f = Take "Foto" 
               c = take Collage
               p =print

with the version 2.0.5 of the photobooth-app every thing is working perfect.

I had some difficulties to get the bluetooth pairing, after I installen blueman on the Rpi it works.

If you need more info please come back.

**Reply by mgineer85** [reply]
Hey, if you want to share the file, maybe you would want to upload it to https://gist.github.com/ and link here for @PixelAlarm

**Reply by mgineer85** [reply]
Sure, just upload the main.ccp and you're done. 
If somebody likes it, he can use it or move forward otherwise.
You could also send a pull-request with your changes and it will be merged upstream - but that is a bit more complicated, so do whatever you like.

**Reply by Striker62** [reply]
I'm not sure what result of the pull request will be. I thing you shold let the original Buzzer as it is . only one button.
My extention should be an alternativ and an example how to add more buttons

**Comment by Striker62** [comment]
I was not sure if you ok with it. 
Never did it before some thing I have to know ?

**Reply by Striker62** [reply]
here it is 
<script src="https://gist.github.com/Striker62/40b16626e0ea539dcdcc195ab8ba02f7.js"></script>
let me know if it is ok that way

**Reply by mgineer85** [reply]
Seems not to work?

**Reply by Striker62** [reply]
the link or the file

**Reply by Striker62** [reply]
for your info I'm ussing ardurion ide to maintain cpp not vscode.

**Reply by Striker62** [reply]
any error report form thr compilation ?

**Reply by Striker62** [reply]
https://www.az-delivery.de/products/esp-32-dev-kit-c-v4

this is the esp I use 

probably you have a different that use an other library for some parts

**Reply by Striker62** [reply]
today I test to install my 3 Button main.cpp again to check if I brake something when I finaly clean up my code.
For me the code works.

But I had again issues with bluetooth in the bullseye version. So I decide to setup from scratch on Bookworm.
Works and no more issues with bluetooth ;-)

**Reply by Striker62** [reply]
any news does my code works for you ?
